createpythonpackage/main.py

```diff
@@ -101,6 +104,12 @@ def _create_package(args):
             }
         )
 
+    if not package_config.package_name.isidentifier():
```

Comment:
I know this is not something the code is encouraging at this time, but this validation error makes the case clear that the error should be raised as soon as a possible, immediately after the user enters it if possible. It would be rather frustrating to go through all the questions only to be told you entered an invalid name.

Ideally there would be an optional validator function that could validate the user's input immediately, the repeat the question until it passed the validator (or the user presses ctrl+c). That is out of the scope of this PR, but just wanted to mention it since this is the perfect use case.

Reply:
Yes, you’re right. I noticed this but as you said I figured this would be out of scope. I’d say we open an issue to refactor the question helper and allow validation afterwards.